crates/agglayer-node/src/service/mod.rs

```diff
@@ -0,0 +1,390 @@
+                hash,
+                "Failed to settle transaction {hash} on L1: {e}"
+            )
+        })?;
+
+        agglayer_telemetry::SETTLE.add(1, metrics_attrs);
+
+        info!(hash, "Successfully settled transaction {hash}");
+
+        Ok(receipt.transaction_hash)
+    }
+
+    #[instrument(skip(self), fields(hash = hash.to_string()), level = "info")]
+    pub async fn get_tx_status(&self, hash: H256) -> Result<TxStatus, TxStatusError<Rpc>> {
+        debug!("Received request to get transaction status for hash {hash}");
+
+        let receipt = self.kernel.check_tx_status(hash).await.map_err(|e| {
+            error!("Failed to get transaction status for hash {hash}: {e}");
+            TxStatusError::StatusCheck(e)
+        })?;
+
+        let current_block = self.kernel.current_l1_block_height().await.map_err(|e| {
+            error!("Failed to get current L1 block: {e}");
+            TxStatusError::L1BlockRetrieval(e)
+        })?;
+
+        let receipt = receipt.ok_or_else(|| TxStatusError::TxNotFound { hash })?;
+
+        let status = match receipt.block_number {
+            Some(block_number) if block_number < current_block => TxStatus::Done,
+            Some(_) => TxStatus::Pending,
+            None => TxStatus::NotFound,
+        };
+        Ok(status)
+    }
+
+    #[instrument(skip(self, certificate), fields(hash, rollup_id = *certificate.network_id), level = "info")]
+    pub async fn send_certificate(
+        &self,
+        certificate: Certificate,
+    ) -> Result<CertificateId, CertificateSubmissionError> {
+        let hash = certificate.hash();
+        tracing::Span::current().record("hash", hash.to_string());
+
+        info!(
+            %hash,
+            "Received certificate {hash} for rollup {} at height {}", *certificate.network_id, certificate.height
+        );
+
+        // TODO: Batch the different queries.
+        // Insert the certificate header into the state store.
+        self.state
+            .insert_certificate_header(&certificate, CertificateStatus::Pending)
+            .inspect_err(|e| error!("Failed to insert certificate into state store: {e}"))?;
+
+        // Insert the certificate into the pending store.
+        self.pending_store
+            .insert_pending_certificate(certificate.network_id, certificate.height, &certificate)
+            .inspect_err(|e| error!("Failed to insert certificate into pending store: {e}"))?;
+
+        self.debug_store
+            .add_certificate(&certificate)
+            .inspect_err(|e| error!("Failed to insert certificate into debug store: {e}"))?;
+
+        self.certificate_sender
+            .send((
+                certificate.network_id,
+                certificate.height,
+                certificate.hash(),
+            ))
+            .await
+            .map_err(|error| {
+                error!("Failed to send certificate: {error}");
+                CertificateSubmissionError::OrchestratorNotResponsive
```

Comment:
I would argue that in general better approach is to keep original errors, just wrap them and bubble up.

Reply:
The `SendError` returned here contains the original sent value which I wanted to get rid of.